public/app/features/templating/TextBoxVariable.ts

```diff
@@ -29,14 +27,72 @@ export class TextBoxVariable implements TextBoxVariableModel, VariableActions {
     options: [],
     skipUrlSync: false,
   };
+  temporary: TextBoxVariableModel = null;
 
   /** @ngInject */
   constructor(private model: any, private variableSrv: VariableSrv) {
-    assignModelProperties(this, model, this.defaults);
+    if (model.useTemporary) {
+      this.temporary = {} as QueryVariableModel;
+      assignModelProperties(this.temporary, model, this.defaults);
+      this.id = -1;
+    } else {
+      this.temporary = null;
+      this.id = createVariableInState(model, this.defaults);
```

Comment:
this part feels a bit wrong, that the constructor is having a redux state side-effect